go/ct/gen/transient_storage_test.go

```diff
@@ -0,0 +1,158 @@
+	rnd := rand.New(0)
+	generator := NewTransientStorageGenerator()
+	assignment := Assignment{}
+	transient, err := generator.Generate(assignment, rnd)
+	if err != nil {
+		t.Errorf("Unexpected error during generation: %v", err)
+	}
+	if transient == nil {
+		t.Errorf("Expected transient storage to be generated, but got nil")
+	}
+	if transient.IsAllZero() {
+		t.Errorf("Expected transient storage to be non-empty, but got empty")
+	}
+}
+
+func TestTransient_ConstraintIsEnforced(t *testing.T) {
+	v1 := Variable("v1")
+	tests := []struct {
+		bind   func(*TransientStorageGenerator)
+		isZero bool
+	}{
+		{
+			bind: func(t *TransientStorageGenerator) {
+				t.BindToNonZero(v1)
+			},
+			isZero: false,
+		},
+		{
+			bind: func(t *TransientStorageGenerator) {
+				t.BindToZero(v1)
+			},
+			isZero: true,
+		},
+	}
+
+	for _, test := range tests {
+		assignment := Assignment{}
+		assignment[v1] = common.NewU256(42)
+
+		rnd := rand.New(0)
+		generator := NewTransientStorageGenerator()
+		test.bind(generator)
+
+		transient, err := generator.Generate(assignment, rnd)
+		if err != nil {
+			t.Errorf("Expected error during generation, %v", err)
+		}
+		if transient.IsZero(common.NewU256(42)) != test.isZero {
+			t.Errorf("Constraint was not enforced")
+		}
+	}
+}
+
+func TestTransient_ConflictingSetUnsetConstraintsAreDetected(t *testing.T) {
+	v1 := Variable("v1")
+	assignment := Assignment{}
+	assignment[v1] = common.NewU256(42)
+
+	rnd := rand.New(0)
+	generator := NewTransientStorageGenerator()
+	generator.BindToNonZero(v1)
+	generator.BindToZero(v1)
+
+	_, err := generator.Generate(assignment, rnd)
+	if err == nil {
+		t.Errorf("Expected error during generation, but got nil")
+	}
+}
+
+func TestTransient_ConstraintsOnTheSameKey(t *testing.T) {
+	v1 := Variable("v1")
+	v2 := Variable("v2")
+	tests := []struct {
+		change  func(*TransientStorageGenerator)
+		success bool
+	}{
+		{
+			change: func(t *TransientStorageGenerator) {
+				t.BindToNonZero(v1)
+				t.BindToNonZero(v2)
+			},
+			success: true,
+		},
+		{
+			change: func(t *TransientStorageGenerator) {
+				t.BindToNonZero(v1)
+				t.BindToZero(v2)
+			},
+			success: false,
+		},
+	}
+
+	for _, test := range tests {
+		assignment := Assignment{}
+		assignment[v1] = common.NewU256(42)
+		assignment[v2] = common.NewU256(42)
+
+		rnd := rand.New(0)
+		generator := NewTransientStorageGenerator()
+
+		test.change(generator)
+
+		_, err := generator.Generate(assignment, rnd)
+		if (err == nil) != test.success {
+			t.Errorf("Constraints on the same key were not enforced correctly")
+		}
+	}
+}
+
+func TestTransient_UnboundVariables(t *testing.T) {
+	v1 := Variable("v1")
+	v2 := Variable("v2")
+	rnd := rand.New(0)
+
+	generator := NewTransientStorageGenerator()
+	generator.BindToNonZero(v1)
+	generator.BindToZero(v2)
+	_, err := generator.Generate(Assignment{}, rnd)
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+}
```

Comment:
This test should also check:
 - that the variables `v1` and `v2` are assigned to storage addresses in the assignment passed to the generator
 - the produced transient storage has a non-zero value at the address associated to `v1` and a zero value associated to `v2`

I do not see this generator outputs be tested in any other test in this file.